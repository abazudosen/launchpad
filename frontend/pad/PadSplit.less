@import '../util.import.less';
@import '../theme.less';

.PadSplit {
  width: 100%;
  height: 100%;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  .wt-container {
    min-width: 0;
  }

  &-Sidebar {
    flex: 0 0 250px;
    padding: 1.5rem 1rem;
    width: 250px;
    border-right: 1px solid #eee;
  }

  &-SidebarItem {
    padding-bottom: 20px;
    margin-bottom: 20px;
    border-bottom: 1px solid fadeout(@colorConcrete, 70%, relative);
  }

  &-Main {
    flex: 1 1;
    display: flex;
    flex-direction: column;
    align-items: stretch;
    position: relative;
  }

  &-Editors {
    flex: 1 1;
    display: flex;
    flex-direction: row;
    position: relative;
  }

  &-Left {
    z-index: 1;
    display: flex;
    flex-direction: column;
    align-items: stretch;

    width: 100%;
    height: 100%;
    position: absolute;
    transition: all 0.2s ease-in;
  }

  &-Right {
    flex: 1 1;
    display: flex;
    overflow-y: auto;
    position: absolute;
    width: 100%;
    height: 100%;
    transition: all 0.2s ease-in;
  }

  &-Hidden {
    flex: 0 1;
  }

  &-EditorWrapper {
    flex: 1 1;
    overflow: auto;
  }

  .editor-fork-banner {
    .font-s1;
    text-align: center;
    background-color: @color-accent-warmer;
    padding: 0.5rem 1rem;
  }

  &-ErrorWrapper {
    // font-size: 0;
    // flex: 0 0 auto;
    // padding: 5px;
    background-color: @color-negative;
    padding: 5px 15px;

    code {
      .font-s1;
      .font-code;
      display: block;
      color: @color-lightest;
    }

    &--yellow {
      background-color: yellow;

      code {
        color: @color-darkest;
      }
    }
  }

  &-ErrorText {
  }

  &-Logs {
    flex: 0 0 auto;
    z-index: 100;
  }
}

.welcome-modal-start-wrapper {
  text-align: center;
  margin-top: 2em;
}

.welcome-modal-start {
  background-color: @color-complementary !important;
}
