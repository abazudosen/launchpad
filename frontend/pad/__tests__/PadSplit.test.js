/* @flow */

import React from 'react';
import PadSplit from '../PadSplit.js';
import ReactShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ReactShallowRenderer();

const testPad = {
  id: 'fakeid1',
  title: 'Fake 1',
  description: 'Description 1',
  code: 'code',
  deployedCode: 'deployed code',
  url: 'http://example.com',
  user: {
    id: 'test-ids',
    githubUsername: 'example',
    pads: [],
  },
  context: [
    {
      key: 'secret',
      value: 'secret',
    },
  ],
  dependencies: [
    {
      name: 'graphql-js',
      version: '0.8.9',
    },
  ],
  defaultQuery: '{ hello }',
  defaultVariables: null,
  token: 'test-token-2',
  draft: null,
};

const testUser = testPad.user;

function renderPadSplit({
  pad,
  user,
  currentCode,
  currentContext,
  isDeploying,
  error,
}) {
  return (
    <PadSplit
      pad={pad}
      user={user}
      engineClient={{}}
      currentCode={currentCode}
      currentContext={currentContext}
      isDeploying={isDeploying}
      error={error}
      onDeploy={() => {}}
      onReset={() => {}}
      onFork={() => {}}
      onCodeChange={() => {}}
      onContextChange={() => {}}
      onLogin={() => {}}
      onLogout={() => {}}
      onSetTitle={() => {}}
      onSetDescription={() => {}}
      onSetDefaultQuery={() => {}}
      onDownload={() => {}}
    />
  );
}

describe('PadSplit', () => {
  test('render normally', () => {
    const component = renderer.render(
      renderPadSplit({
        pad: testPad,
        user: testUser,
        currentCode: 'test',
        currentContext: [{ key: 'test', value: 'testValue' }],
        isDeploying: false,
        error: null,
      }),
    );

    expect(component).toMatchSnapshot();
  });
});
