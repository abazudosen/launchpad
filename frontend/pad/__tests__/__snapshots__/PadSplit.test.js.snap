// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render normally 1`] = `
<div
  className="PadSplit"
>
  <Header
    currentContext={
      Array [
        Object {
          "key": "test",
          "value": "testValue",
        },
      ]
    }
    engineClient={Object {}}
    isDeploying={false}
    onContextChange={[Function]}
    onDeploy={[Function]}
    onDownload={[Function]}
    onFork={[Function]}
    onLogin={[Function]}
    onLogout={[Function]}
    onReset={[Function]}
    onSetDescription={[Function]}
    onSetTitle={[Function]}
    pad={
      Object {
        "code": "code",
        "context": Array [
          Object {
            "key": "secret",
            "value": "secret",
          },
        ],
        "defaultQuery": "{ hello }",
        "defaultVariables": null,
        "dependencies": Array [
          Object {
            "name": "graphql-js",
            "version": "0.8.9",
          },
        ],
        "deployedCode": "deployed code",
        "description": "Description 1",
        "draft": null,
        "id": "fakeid1",
        "title": "Fake 1",
        "token": "test-token-2",
        "url": "http://example.com",
        "user": Object {
          "githubUsername": "example",
          "id": "test-ids",
          "pads": Array [],
        },
      }
    }
    user={
      Object {
        "githubUsername": "example",
        "id": "test-ids",
        "pads": Array [],
      }
    }
  />
  <div
    className="PadSplit-Main"
  >
    <div
      className="PadSplit-Editors"
    >
      <SplitPane
        allowResize={true}
        defaultSize="50%"
        minSize={50}
        prefixer={
          Prefixer {
            "_browserInfo": Object {
              "blink": true,
              "browserName": "safari",
              "browserVersion": NaN,
              "cssPrefix": "-webkit-",
              "jsPrefix": "Webkit",
              "linux": true,
              "name": "Safari",
              "osVersion": NaN,
              "safari": true,
              "x": true,
            },
            "_hasPropsRequiringPrefix": false,
            "_keepUnprefixed": false,
            "_metaData": Object {
              "browserName": "safari",
              "browserVersion": NaN,
              "cssPrefix": "-webkit-",
              "jsPrefix": "Webkit",
              "keepUnprefixed": false,
              "requiresPrefix": Object {},
            },
            "_requiresPrefix": Object {},
            "_userAgent": "Node.js (linux; U; rv:v6.11.2) AppleWebKit/537.36 (KHTML, like Gecko)",
            "prefixedKeyframes": "keyframes",
          }
        }
        primary="first"
        split="vertical"
      >
        <div
          className="PadSplit-Left"
        >
          <div
            className="PadSplit-EditorWrapper"
          >
            <Editor
              canEdit={true}
              code="test"
              onChange={[Function]}
            />
          </div>
          <div
            className="PadSplit-Logs"
          />
        </div>
        <div
          className="PadSplit-Right"
        >
          <GraphiQLWrapper
            isDeploying={false}
            onSetDefaultQuery={[Function]}
            pad={
              Object {
                "code": "code",
                "context": Array [
                  Object {
                    "key": "secret",
                    "value": "secret",
                  },
                ],
                "defaultQuery": "{ hello }",
                "defaultVariables": null,
                "dependencies": Array [
                  Object {
                    "name": "graphql-js",
                    "version": "0.8.9",
                  },
                ],
                "deployedCode": "deployed code",
                "description": "Description 1",
                "draft": null,
                "id": "fakeid1",
                "title": "Fake 1",
                "token": "test-token-2",
                "url": "http://example.com",
                "user": Object {
                  "githubUsername": "example",
                  "id": "test-ids",
                  "pads": Array [],
                },
              }
            }
            user={
              Object {
                "githubUsername": "example",
                "id": "test-ids",
                "pads": Array [],
              }
            }
          />
        </div>
      </SplitPane>
    </div>
  </div>
  <Footer
    canSeeLogs={true}
    error={null}
    isDeployed={undefined}
    isDraft={undefined}
    isLogOpen={false}
    onLogClose={[Function]}
    onLogOpen={[Function]}
    onModalOpen={[Function]}
    onResetLinkClick={[Function]}
    url="http://example.com"
  />
  <Modal
    isOpen={true}
    onRequestClose={[Function]}
    title="Welcome to Apollo Launchpad!"
  >
    <p>
      This is a tool you can use build, deploy, and share a simple GraphQL API right from your browser. We think it's a great way to experiment with GraphQL and share examples of different patterns.
    </p>
    <p>
      Visit the
       
      <a
        href="https://github.com/apollographql/awesome-launchpad"
        target="_blank"
      >
        awesome-launchpad repository
      </a>
       
      to see a list of examples and read the docs. For an introduction, read the
       
      <a
        href="https://dev-blog.apollodata.com/introducing-launchpad-the-graphql-server-demo-platform-cc4e7481fcba"
        target="_blank"
      >
        announcement blog post
      </a>
      .
    </p>
    <div
      className="welcome-modal-start-wrapper"
    >
      <button
        className="welcome-modal-start btn primary"
        onClick={[Function]}
      >
        Start using Launchpad
      </button>
    </div>
  </Modal>
  <Modal
    isOpen={false}
    onRequestClose={[Function]}
    title="Edit Server Secrets"
  >
    <ContextEditor
      context={
        Array [
          Object {
            "key": "test",
            "value": "testValue",
          },
        ]
      }
      onChange={[Function]}
    />
  </Modal>
  <Modal
    isOpen={false}
    onRequestClose={[Function]}
    title="npm Dependencies"
  >
    <Dependencies
      dependencies={
        Array [
          Object {
            "name": "graphql-js",
            "version": "0.8.9",
          },
        ]
      }
    />
  </Modal>
</div>
`;
